src/microbe_stage/MicrobeAI.cs

```diff
@@ -65,7 +85,14 @@ public void Think(float delta, Random random, MicrobeAICommonData data)
 
         ChooseActions(random, data);
 
-        // Clear the absorbed compounds for run and rumble
+        // Store the absorbed compounds for run and rumble
+        previouslyAbsorbedCompounds.Clear();
+        foreach (var compound in microbe.TotalAbsorbedCompounds)
+        {
+            previouslyAbsorbedCompounds[compound.Key] = compound.Value;
+        }
```

Comment:
```suggestion
        previouslyAbsorbedCompounds = new Dictionary<Compound, float>(microbe.TotalAbsorbedCompounds);
```
This also makes a copy of the dictionary

Reply:
This was specifically not done to not create a new dictionnary instance.